src/SFA.DAS.PR.Application/Requests/Commands/AcceptCreateAccountRequest/AcceptCreateAccountRequestCommandHandler.cs

```diff
@@ -0,0 +1,199 @@
+using SFA.DAS.ProviderRelationships.Messages.Events;
+using SFA.DAS.ProviderRelationships.Types.Models;
+
+namespace SFA.DAS.PR.Application.Requests.Commands.AcceptCreateAccountRequest;
+
+public sealed class AcceptCreateAccountRequestCommandHandler(
+    ILogger<AcceptCreateAccountRequestCommandHandler> _logger,
+    IEncodingService _encodingService,
+    IProviderRelationshipsDataContext _providerRelationshipsDataContext,
+    IAccountWriteRepository _accountWriteRepository,
+    IAccountProviderWriteRepository _accountProviderWriteRepository,
+    IAccountLegalEntityWriteRepository _accountLegalEntityWriteRepository,
+    IAccountProviderLegalEntitiesWriteRepository _accountProviderLegalEntitiesWriteRepository,
+    IPermissionsWriteRepository _permissionsWriteRepository,
+    IPermissionsAuditWriteRepository _permissionsAuditWriteRepository,
+    IMessageSession _messageSession,
+    IRequestWriteRepository _requestWriteRepository
+) : IRequestHandler<AcceptCreateAccountRequestCommand, ValidatedResponse<Unit>>
+{
+    public async Task<ValidatedResponse<Unit>> Handle(AcceptCreateAccountRequestCommand command, CancellationToken cancellationToken)
+    {
+        await CreateAccountAndAddPermissions(command, cancellationToken);
+
+        return ValidatedResponse<Unit>.EmptySuccessResponse();
+    }
+
+    private async Task CreateAccountAndAddPermissions(AcceptCreateAccountRequestCommand command, CancellationToken cancellationToken)
+    {
+        Request? request = await _requestWriteRepository.GetRequest(command.RequestId, cancellationToken);
+
+        await CreateAccount(command, cancellationToken);
+
+        await CreateAccountLegalEntity(command, cancellationToken);
+
+        Tuple<bool, AccountProvider> providerResponse = await GetOrCreateAccountProvider(command, request!, cancellationToken);
+            
+        AccountProviderLegalEntity accountProviderLegalEntity = 
+            await _accountProviderLegalEntitiesWriteRepository.CreateAccountProviderLegalEntity(
+                command.AccountLegalEntity.Id,
+                providerResponse.Item2,
+                cancellationToken
+        );
+
+        IEnumerable<Permission> permissions = request!.PermissionRequests.Select(pr => new Permission()
+        {
+            AccountProviderLegalEntity = accountProviderLegalEntity,
+            Operation = (Operation)pr.Operation
+        });
+
+        _permissionsWriteRepository.CreatePermissions(permissions);
+
+        await CreatePermissionsAudit(
+            command,
+            request,
+            permissions.Select(a => a.Operation),
+            RequestAction.AccountCreated,
+            cancellationToken
+        );
+
+        request.ActionedBy = command.ActionedBy;
+        request.Status = RequestStatus.Accepted;
+        request.UpdatedDate = DateTime.UtcNow;
+
+        await _providerRelationshipsDataContext.SaveChangesAsync(cancellationToken);
+
+        if (providerResponse.Item1)
```

Comment:
'Item1' is very cryptic and unreadable. Please introduce a variable like 'var IsGetOrCreateProvicerSuccessful = providerResponse.Item1' and use that in the If.  It's easier to read what's going on.